File: backend/ai_modules/fairness_metrics.py
"""def calculate_fairness(male, female):

    if male + female == 0:
        return 0

    score = min(male, female) / max(male, female)

    return score


if __name__ == "__main__":
    print(calculate_fairness(6,4))"""

def compute_fairness(disparate_impact):

    if 0.8 <= disparate_impact <= 1.25:
        fairness = "Fair Hiring"
    else:
        fairness = "Bias Detected"

    return {
        "fairness_status": fairness
    }

File: frontend/src/components/FairnessMetrics.js
import React, { useEffect, useState } from "react";
import axios from "axios";

function FairnessMetrics() {

  const [bias, setBias] = useState({});

const fetchBias = async () => {

  try {
    const res = await axios.get("http://localhost:8000/bias_report");
    setBias(res.data);
  } catch (error) {

    if (error.response) {
      alert(error.response.data.detail);
    } else {
      alert("Server error while checking bias");
    }

  }

};

  useEffect(() => {
   
    fetchBias();
  }, []);

  if (!bias || !bias.bias_metrics) {
  return <p>Resumes Upload Successfully</p>;
}

  return (
    <div>
      <h2>Fairness Metrics</h2>

      <p>Male Selection Rate: {bias.bias_metrics.male_selection_rate}</p>
      <p>Female Selection Rate: {bias.bias_metrics.female_selection_rate}</p>
      <p>Disparate Impact: {bias.bias_metrics.disparate_impact}</p>

      <h3>{bias.fairness.fairness_status}</h3>
    </div>
  );
}

export default FairnessMetrics;